var/spack/repos/builtin/packages/hpx/package.py

```diff
@@ -279,12 +279,15 @@ def cmake_args(self):
 
         if spec.satisfies("instrumentation=apex"):
             args += [
+                self.define("HPX_WITH_FETCH_APEX", True),
```

Comment:
Can't we depend on `apex` and point to the Spack installed package? Fetching a dependency like this means it's not managed by Spack...

Reply:
I fully agree. From a quick look at [HPX CMake](https://github.com/STEllAR-GROUP/hpx/blob/b60bcf235d509e14a6979b3977b1def05d41d9c1/CMakeLists.txt#L994-L1001) it is possible to both fetch or use an external one, so it is possible to mimic what [`pika`](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/pika/package.py#L95) does (look for `apex` variant).

Reply:
I looked at pika and HPX uses the spack package for apex now.